### ISSUE TITLE
better error instead of "error at replicate"

### ISSUE THREAD

**Issue by githubmanticore** [nck-2/test-rep#430]
TL;DR when you make an INSERT into a replicated table on a node which is being a donor there's a high chance to get 
``` 
ERROR 1064 (42000) at line 1: error at replicate, code 3, seqno -1 
``` 
 
If it's a normal situation and we can't just wait until the node is done with being a donor let's make the error less confusing, for example  
``` 
Replication error: the node is busy, can't complete the transaction, try again 
``` 
 
Here's how to reproduce it (I did it a few times on our dev server): 
 
# Node 1 
 
Empty the `data_dir` and run searchd with this config: 
 
``` 
searchd { 
    listen = 127.0.0.1:10201 
    listen = 127.0.0.1:10301:mysql 
    log = searchd.log 
    query_log = query.log 
    pid_file = searchd.pid 
    data_dir = data 
    binlog_path = 
} 
``` 
 
# Node 2 
 
Empty the `data_dir` and run searchd with this config: 
 
``` 
searchd { 
    listen = 127.0.0.1:20201 
    listen = 127.0.0.1:20301:mysql 
    log = searchd.log 
    query_log = query.log 
    pid_file = searchd.pid 
    data_dir = data 
    binlog_path = 
} 
``` 
 
# Node 1 
 
Initialize a cluster and a table: 
 
```bash 
port=10301;  
mysql -P$port -f -h0 -e "create cluster c;" 
s=""; stringLength=20; batchSize=3000; queriesCount=3000; for n in `seq 1 $stringLength`; do s="a$s"; done; mysql -P$port -h0 -e "drop table if exists t; create table t(f text, a int)"; q="insert into t values "; for n in `seq 1 $batchSize`; do q="$q (0,'$s',34324324324343)"; [ $n != $batchSize ] && q="$q,"; done; q="$q;"; for m in `seq 1 $queriesCount`; do mysql -P$port -h0 -e "$q"; echo -n "."; done; 
mysql -P$port -f -h0 -e "alter cluster c add t;" 
``` 
 
Prepare this query to be run: 
 
```bash 
s=""; stringLength=10; batchSize=3000; queriesCount=30; for n in `seq 1 $stringLength`; do s="a$s"; done; q="insert into c:t values "; for n in `seq 1 $batchSize`; do q="$q (0,'$s',34324324324343)"; [ $n != $batchSize ] && q="$q,"; done; q="$q;"; for m in `seq 1 $queriesCount`; do date; (time mysql -P$port -h0 -e "$q") 2>&1|egrep -v "user|sys"; done; 
``` 
 
# Node 2 
 
Prepare this query to be run: 
 
```bash 
date; time mysql -P20301 -h0 -e "join cluster c at '127.0.0.1:10201'" 
``` 
 
# Run 
 
Hit Enter on the node 1, wait until the inserts start and instantly hit Enter on the node 2 
 
You should get smth like this: 
 
``` 
Sun Oct 23 08:58:31 CEST 2022 
 
real    0m0.039s 
Sun Oct 23 08:58:31 CEST 2022 
 
real    0m0.042s 
Sun Oct 23 08:58:31 CEST 2022 
ERROR 1064 (42000) at line 1: error at replicate, code 3, seqno -1 
 
real    0m0.521s 
Sun Oct 23 08:58:32 CEST 2022 
 
real    0m3.922s 
``` 
 
Sometimes it requires a few attempts.
